vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc

```diff
@@ -0,0 +1,203 @@
+
+#include "bat/ledger/internal/bat_helper.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/rapidjson_bat_helper.h"
+#include "bat/ledger/internal/wallet/create.h"
+#include "bat/ledger/internal/wallet/recover.h"
+#include "net/http/http_status_code.h"
+
+#include "wally_bip39.h"  // NOLINT
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+using std::placeholders::_3;
+
+namespace braveledger_wallet {
+
+Wallet::Wallet(bat_ledger::LedgerImpl* ledger) :
+    ledger_(ledger),
+    create_(std::make_unique<Create>(ledger)),
+    recover_(std::make_unique<Recover>(ledger)) {
+}
+
+Wallet::~Wallet() {
+}
+
+void Wallet::CreateWalletIfNecessary() {
+  const auto payment_id = ledger_->GetPaymentId();
+  const auto stamp = ledger_->GetBootStamp();
+  const auto persona_id = ledger_->GetPersonaId();
+
+  if (!payment_id.empty() && stamp != 0 && !persona_id.empty()) {
+    ledger_->OnWalletInitialized(ledger::Result::WALLET_CREATED);
+    return;
+  }
+
+  BLOG(ledger_, ledger::LogLevel::LOG_ERROR) <<
+     "Wallet creation didn't finish or corrupted. " <<
+     "We need to clear persona Id and start again";
+  ledger_->SetPersonaId("");
+
+  create_->Start();
+}
+
+void Wallet::GetWalletProperties(
+    ledger::OnWalletPropertiesCallback callback) {
+  std::string payment_id = ledger_->GetPaymentId();
+  std::string passphrase = GetWalletPassphrase();
+
+  if (payment_id.empty() || passphrase.empty()) {
+    braveledger_bat_helper::WALLET_PROPERTIES_ST properties;
+    ledger_->OnWalletProperties(ledger::Result::CORRUPTED_WALLET, properties);
+    return;
+  }
+
+  std::string path = (std::string)WALLET_PROPERTIES
+      + payment_id
+      + WALLET_PROPERTIES_END;
+  const std::string url = braveledger_bat_helper::buildURL(
+      path,
+      PREFIX_V2,
+      braveledger_bat_helper::SERVER_TYPES::BALANCE);
+  auto load_callback = std::bind(&Wallet::WalletPropertiesCallback,
+                            this,
+                            _1,
+                            _2,
+                            _3,
+                            callback);
+  ledger_->LoadURL(url,
+                   std::vector<std::string>(),
+                   std::string(),
+                   std::string(),
+                   ledger::URL_METHOD::GET,
+                   load_callback);
+}
+
+ledger::WalletInfo Wallet::WalletPropertiesToWalletInfo(
+    const braveledger_bat_helper::WALLET_PROPERTIES_ST& properties) {
+  ledger::WalletInfo info;
+  info.altcurrency_ = properties.altcurrency_;
+  info.probi_ = properties.probi_;
+  info.balance_ = properties.balance_;
+  info.rates_ = properties.rates_;
+  info.parameters_choices_ = properties.parameters_choices_;
+  info.fee_amount_ = ledger_->GetContributionAmount();
+  info.parameters_range_ = properties.parameters_range_;
+  info.parameters_days_ = properties.parameters_days_;
+
+  for (size_t i = 0; i < properties.grants_.size(); i ++) {
```

Comment:
Would use an iterator here if possible, unless you're trying to avoid any actual functional changes in this PR.

Reply:
yeah I want to avoid changing any functionality

Reply:
OK, no worries.